src/components/task-plan/homework.cjsx

```diff
@@ -0,0 +1,404 @@
+
+  renderChapterPanels: (chapter, i) ->
+    sections = _.map(chapter.children, @renderSections)
+    header =
+      <h2 className="-chapter-title">
+        <span className="-chapter-number">{chapter.number}</span>
+        <span className="-chapter-title">{chapter.title}</span>
+      </h2>
+
+    <BS.Accordion>
+      <BS.Panel key={chapter.id} header={header} eventKey={chapter.id}>
+        {sections}
+      </BS.Panel>
+    </BS.Accordion>
+
+  selectProblems: ->
+    @setState({
+      showProblems: true
+    })
+
+  render: ->
+    {courseId, planId, selected, hide} = @props
+    unless @props.shouldShow
+      return <span className="-no-show"/>
+    unless TocStore.isLoaded()
+      TocActions.load(courseId)
+      return <span className="-loading">Loading...</span>
+
+    chapters = _.map(TocStore.get(), @renderChapterPanels)
+    buttonStyle = if selected?.length then 'primary' else 'disabled'
+    shouldShowExercises = selected?.length and @state.showProblems
+
+    if not @state.showProblems
+      footer =
+        <span>
+          <BS.Button bsStyle={buttonStyle} onClick={@selectProblems}>Show Problems</BS.Button>
+          <BS.Button bsStyle="link" onClick={hide}>Cancel</BS.Button>
+        </span>
+
+    <div>
+      <BS.Panel footer={footer}>
+        <div className="select-reading-modal">
+          {chapters}
+        </div>
+      </BS.Panel>
+      <ExerciseSummary
+        shouldShow={shouldShowExercises} 
+        canReview={true}
+        reviewClicked={hide}
+        planId={planId}/>
+      <AddExercises
+        courseId={courseId} 
+        planId={planId} 
+        shouldShow={shouldShowExercises} 
+        pageIds={selected}
+        hide={@hideSectionTopics}/>
+    </div>
+
+HomeworkPlan = React.createClass
+  getInitialState: ->
+    {showSectionTopics: false}
+
+  setDueAt: (value) ->
+    {id} = @props
+    TaskPlanActions.updateDueAt(id, value)
+
+  setTitle: ->
+    {id} = @props
+    value = @refs.title.getDOMNode().value
+    TaskPlanActions.updateTitle(id, value)
+
+  setDescription: ->
+    {id} = @props
+    desc = @refs.description.getDOMNode().value
+    TaskPlanActions.updateDescription(id, desc)
+  
+  showSectionTopics: ->
+    @setState({
+      showSectionTopics: true
+    })
+
+  hideSectionTopics: ->
+    @setState({
+      showSectionTopics: false
+    })
+
+  render: ->
+    {id, courseId} = @props
+    plan = TaskPlanStore.get(id)
+    description = TaskPlanStore.getDescription(id)
+    headerText = if TaskPlanStore.isNew(id) then 'Add Homework' else 'Edit Homework'
+    topics = TaskPlanStore.getTopics(id)
+    shouldShowExercises = TaskPlanStore.getExercises(id)?.length and not @state?.showSectionTopics
+
+    if plan?.due_at
+      dueAt = new Date(plan.due_at)
+
+    today = new Date(plan.due_at)
+    footer = <PlanFooter id={id} courseId={courseId} clickedSelectProblem={@showSectionTopics}/>
+
+    formClasses = ['create-homework']
```

Comment:
This should probably be `edit-homework` since the class occurs both when creating and when editing a HW.